package/MDAnalysis/analysis/distances.py

```diff
@@ -120,7 +124,7 @@ def contact_matrix(coord, cutoff=15.0, returntype="numpy", box=None):
         return sparse_contacts
 
 
-def dist(A, B, offset=0, box=None):
+def dist(A: AtomGroup, B: AtomGroup, offset: Union[int, Tuple] = 0, box=None) -> np.ndarray:
```

Comment:
Here, `offset` is either a single `int` or 2 `int`s. In the latter case, it is used here: https://github.com/MDAnalysis/mdanalysis/blob/5e8d5f6f957418e3ca74812b6c48870c78e0bbd5/package/MDAnalysis/analysis/distances.py#L165

The strict annotation would be `Union[int, Tuple[int, int]]`. This guarantees the use of the `offset` argument cannot fail. However, it also forbids some valid cases. While it authorises `offset=2` and `offset=(3, 4)`, it does not let a user do `offset=[3, 4]` or `offset=np.array(3, 4)` which is valid in python.

A broader annotation is `Union[int, Iterable[int]]`. This does not offer the same guarantees as the stricter annotation, but it does allow everything valid. Indeed, the syntax `a, b = c` is called unpacking, and it only requires `c` to be [iterable](https://docs.python.org/3/glossary.html#term-iterable). @fiona-naughton explained what it meant on her blog a while ago: https://fiona-naughton.github.io/blog/2016/06/08/How-to-train-your-Python.